providers/openai/errors.go

```
@@ -35,3 +39,67 @@ func apiErrorMessage(apiErr *openaisdk.Error) string {
+// apierror.Error" and no longer classifies for retry.
+//
+// Rewriting the body into the shape the SDK expects keeps the status code, the
+// message, and the retry classification intact, and leaves standard-shaped
+// responses untouched.
+func normalizeErrorBodyMiddleware(req *http.Request, next option.MiddlewareNext) (*http.Response, error) {
+	res, err := next(req)
+	if err != nil || res == nil || res.StatusCode < 400 || res.Body == nil {
+		return res, err
+	}
+	body, readErr := io.ReadAll(res.Body)
+	res.Body.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `res.Body.Close` is not checked

(errcheck)